cypress/page-objects/paymentModal.js

```diff
@@ -0,0 +1,255 @@
+    return cy.iframe().find('span').contains('Billing');
+  }
+
+  closePaymentModal() {
+    this.closeModalCTA.should('be.visible');
+    this.closeModalCTA.click();
+  }
+
+  switchToPaymentModal() {
+    cy.frameLoaded('.stripe_checkout_app');
+  }
+
+  getTotalPrice() {
+    return this.purchaseDescription.invoke('text')
+      .then((text) => _.toNumber(text.match(/(\d*$)/)[0]));
+  }
+
+  enterEmail(emailAddress) {
+    this.emailField.type(emailAddress);
+  }
+
+  enterShippingName(shippingName) {
+    this.shippingNameField.type(shippingName);
+  }
+
+  enterShippingStreet(shippingStreet) {
+    this.shippingStreetField.type(shippingStreet);
+  }
+
+  enterShippingPostcode(shippingPostcode) {
+    this.shippingPostcodeField.type(shippingPostcode);
+  }
+
+  selectShippingCountry(shippingCountry) {
+    this.shippingCountryDropdown.select(shippingCountry);
+  }
+
+  proceedToPaymentInfoForm() {
+    this.paymentInfoCTA.click();
+  }
+
+  enterCardNumber(cardNumber) {
+    this.cardNumberField.type(cardNumber);
+  }
+
+  enterCardExpirationDate(cardExpirationDate) {
+    this.cardExpirationDateField.type(cardExpirationDate);
+  }
+
+  enterCardCVC(cardCVC) {
+    this.CVCField.type(cardCVC);
+  }
+
+  navigateBackToPaymentInfoForm() {
+    this.navBackToPaymentInfoFormArrow.should('be.visible');
+    this.navBackToPaymentInfoFormArrow.click();
+  }
+
+  submitPurchase() {
+    this.payNowCTA.click();
+  }
+
+  uncheckSameBillingShippingInfoCheckBox() {
+    this.sameBillingAndShippingInfoCheckBox.click();
+  }
+
+  enterBillingName(billingName) {
+    this.billingNameField.type(billingName);
+  }
+
+  enterBillingStreet(billingStreet) {
+    this.billingStreetField.type(billingStreet);
+  }
+
+  enterBillingCity(billingCity) {
+    this.billingCityField.type(billingCity);
+  }
+
+  enterBillingPostcode(billingPostcode) {
+    this.billingPostcodeField.type(billingPostcode);
+  }
+
+  selectBillingCountry(billingCountry) {
+    this.billingCountryDropdown.select(billingCountry);
+  }
+
+  switchToBillingInfoSection() {
+    this.billingSectionToggle.click();
+  }
+
+  enterShippingInformation(
+    emailAddress,
+    shippingName,
+    shippingCountry,
+    shippingStreet,
+    shippingPostcode,
+    shippingCity,
+    isSimilarToPaymentInformation = true,
+  ) {
+    this.shippingNameField.should('be.visible');
+    this.selectShippingCountry(shippingCountry);
+    if (emailAddress) {
+      this.enterEmail(emailAddress);
+    }
+    if (shippingName) {
+      this.enterShippingName(shippingName);
+    }
+    if (shippingStreet) {
+      this.enterShippingStreet(shippingStreet);
+    }
+    if (shippingPostcode) {
+      this.enterShippingPostcode(shippingPostcode);
+    }
+    if (shippingCity) {
+      cy.iframe().find('#shipping-city').clear().type(shippingCity);
+    } else {
+      cy.iframe().find('#shipping-city').clear();
+    }
+    if (isSimilarToPaymentInformation) {
+      this.proceedToPaymentInfoForm();
+    }
+  }
```

Comment:
it is hard to read in specs while passing too many args

```
// you could pass your args as an object

enterShippingInformation(shippingInformation) { ... }

// data object would look like 

shippingInformation = {
    emailAddress: "bla-bla-bla",
    shippingName: "bla-bla-bla",
    shippingCountry: "bla-bla-bla",
    shippingStreet: "bla-bla-bla",
    shippingPostcode: "bla-bla-bla",
    shippingCity: "bla-bla-bla",
    isSimilarToPaymentInformation: true,
}

// and it might be generated as

const shippingInformation = getShippingInformation()

// and generator could look like:

const getShippingInformation = () => {
  return {
    emailAddress: "bla-bla-bla",
    shippingName: "bla-bla-bla",
    shippingCountry: "bla-bla-bla",
    shippingStreet: "bla-bla-bla",
    shippingPostcode: "bla-bla-bla",
    shippingCity: "bla-bla-bla",
    isSimilarToPaymentInformation: true,
  }
}

// and in case of required of empty or invalid values you can just assign that value to your generated object like
// for example, you need data with empty email

const shippingInformation = getShippingInformation()
shippingInformation.emailAddress = null
enterShippingInformation(shippingInformation)
```